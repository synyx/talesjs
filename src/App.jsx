import React from "react";
import ImagePicker from "./components/ImagePicker";
import Stage from "./components/Stage";

export default class App extends React.Component {
  state = {
    image: null,
  };

  handleFileSelect(file) {
    const reader = new FileReader();
    reader.onload = event => {
      this.setState({ image: { src: event.target.result } });
    };
    reader.readAsDataURL(file);
  }

  render() {
    const image = this.state.image;
    return (
      <React.Fragment>
        <ImagePicker onFileSelect={data => this.handleFileSelect(data)} />
        <Stage image={image} />
      </React.Fragment>
    );
  }
}
