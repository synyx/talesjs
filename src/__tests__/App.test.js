import React from "react";
import { renderShallow } from "../../test";
import App from "../App";

describe("App", () => {
  it("renders", () => {
    const tree = renderShallow(<App />);
    expect(tree).toMatchSnapshot();
  });
});
