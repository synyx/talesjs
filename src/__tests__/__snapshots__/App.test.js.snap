// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<React.Fragment>
  <ImagePicker
    onFileSelect={[Function]}
  />
  <Stage
    image={null}
  />
</React.Fragment>
`;
